core/src/main/java/com/gdx/game/screen/OptionScreen.java

```diff
@@ -59,33 +82,234 @@ private void loadContents() {
 
-        Actor controlButton = optionTable.getCells().get(0).getActor();
+        Actor controlButton = optionTable.getCells().get(optionTable.getCells().size - 1).getActor();
+
         controlButton.addListener(new ClickListener() {
             @Override
             public void clicked(InputEvent even, float x, float y) {
                 controlClickListener = true;
+
                 controlTable = createTable();
+                handleControlSettings();
                 handleControlBackButton();
             }
         });
     }
 
+    private void handleControlSettings() {
+
+        Json jsonObject = new Json();
+
+        try {
+            playerControlsNew = jsonObject.fromJson(HashMap.class, Gdx.files.internal(PARTIAL_CONTROLS_SETTINGS_PATH));
+
+            if (DEFAULT_CONTROLS.size() != playerControlsNew.size()){
+                throw new SerializationException("Not valid control map");
+            }
+
+        }catch (SerializationException se){
+
+            LOGGER.error(se.getMessage());
+
+            playerControlsNew = DEFAULT_CONTROLS;
+        }
+
+
+        playerControlsNew = mapInverter(playerControlsNew);
+
+        Label upLabel = new Label(InputComponent.Keys.UP.name(), skin);
+        TextField upText = new TextField(
+                Input.Keys.toString(Integer.parseInt(playerControlsNew.get(InputComponent.Keys.UP.name()))), skin);
+        Label downLabel = new Label(InputComponent.Keys.DOWN.name(), skin);
+        TextField downText = new TextField(
+                Input.Keys.toString(Integer.parseInt(playerControlsNew.get(InputComponent.Keys.DOWN.name()))), skin);
+        Label leftLabel = new Label(InputComponent.Keys.LEFT.name(), skin);
+        TextField leftText = new TextField(
+                Input.Keys.toString(Integer.parseInt(playerControlsNew.get(InputComponent.Keys.LEFT.name()))), skin);
+        Label rightLabel = new Label(InputComponent.Keys.RIGHT.name(), skin);
+        TextField rightText = new TextField(
+                Input.Keys.toString(Integer.parseInt(playerControlsNew.get(InputComponent.Keys.RIGHT.name()))), skin);
+        Label interactLabel = new Label(InputComponent.Keys.INTERACT.name(), skin);
+        TextField interactText = new TextField(
+                Input.Keys.toString(Integer.parseInt(playerControlsNew.get(InputComponent.Keys.INTERACT.name()))), skin);
+        Label optionLabel = new Label(InputComponent.Keys.OPTION.name(), skin);
+        TextField optionText = new TextField(
+                Input.Keys.toString(Integer.parseInt(playerControlsNew.get(InputComponent.Keys.OPTION.name()))), skin);
+        Label quitLabel = new Label(InputComponent.Keys.QUIT.name(), skin);
+        TextField quitText = new TextField(
+                Input.Keys.toString(Integer.parseInt(playerControlsNew.get(InputComponent.Keys.QUIT.name()))), skin);
+
+        playerControlsNew = mapInverter(playerControlsNew);
```

Comment:
what is the use of this mapInverter and the one line 129?
do you need the two of them?